internal/server/middleware/grpc/middleware.go

```diff
@@ -88,33 +90,127 @@ type ResponseDurationRecordable interface {
-	startTime := time.Now().UTC()
+func EvaluationUnaryInterceptor(analyticsEnabled bool) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, _ *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp interface{}, err error) {
+		startTime := time.Now().UTC()
 
-	// set request ID if not present
-	requestID := uuid.Must(uuid.NewV4()).String()
-	if r, ok := req.(RequestIdentifiable); ok {
-		requestID = r.SetRequestIDIfNotBlank(requestID)
+		// set request ID if not present
+		requestID := uuid.Must(uuid.NewV4()).String()
+		if r, ok := req.(RequestIdentifiable); ok {
+			requestID = r.SetRequestIDIfNotBlank(requestID)
 
-		resp, err = handler(ctx, req)
-		if err != nil {
-			return resp, err
-		}
+			resp, err = handler(ctx, req)
+			if err != nil {
+				return resp, err
+			}
 
-		// set request ID on response
-		if r, ok := resp.(RequestIdentifiable); ok {
-			_ = r.SetRequestIDIfNotBlank(requestID)
-		}
+			// set request ID on response
+			if r, ok := resp.(RequestIdentifiable); ok {
+				_ = r.SetRequestIDIfNotBlank(requestID)
+			}
 
-		// record start, end, duration on response types
-		if r, ok := resp.(ResponseDurationRecordable); ok {
-			r.SetTimestamps(startTime, time.Now().UTC())
+			// record start, end, duration on response types
+			if r, ok := resp.(ResponseDurationRecordable); ok {
+				r.SetTimestamps(startTime, time.Now().UTC())
+			}
+
+			if analyticsEnabled {
+				span := trace.SpanFromContext(ctx)
+
+				switch r := resp.(type) {
+				case *evaluation.VariantEvaluationResponse:
+					// This "should" always be an evalution request under these circumstances.
+					if evaluationRequest, ok := req.(*evaluation.EvaluationRequest); ok {
+						evaluationResponses := []*analytics.EvaluationResponse{
+							{
+								NamespaceKey: evaluationRequest.GetNamespaceKey(),
+								FlagKey:      r.GetFlagKey(),
+								FlagType:     evaluation.EvaluationFlagType_VARIANT_FLAG_TYPE.String(),
+								Match:        r.GetMatch(),
+								Reason:       r.GetReason().String(),
+								Timestamp:    r.GetTimestamp().AsTime(),
+							},
+						}
+
+						if evaluationResponsesBytes, err := json.Marshal(evaluationResponses); err == nil {
+							keyValue := []attribute.KeyValue{
+								{
+									Key:   "flipt.evaluation.response",
+									Value: attribute.StringValue(string(evaluationResponsesBytes)),
+								},
+							}
+							span.AddEvent("evaluation_response", trace.WithAttributes(keyValue...))
+						}
+					}
+				case *evaluation.BooleanEvaluationResponse:
+					if evaluationRequest, ok := req.(*evaluation.EvaluationRequest); ok {
+						evaluationResponses := []*analytics.EvaluationResponse{
+							{
+								NamespaceKey: evaluationRequest.GetNamespaceKey(),
+								FlagKey:      r.GetFlagKey(),
+								FlagType:     evaluation.EvaluationFlagType_BOOLEAN_FLAG_TYPE.String(),
+								Match:        r.GetEnabled(),
```

Comment:
Im wondering if we should **not** set this field for boolean type flag.
Instead, I think we should add a `value` to evaluation response.

Variant can provide the `match` property, but it will set `value` to the matched variant.
Boolean will leave `match` unset, but set `value` to `enabled`.

Reply:
@GeorgeMac Ahh this actually makes more sense! I forgot that Boolean's enabled property is the `value` itself